plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,489 @@
+    return self._redis_client.hlen(container_key)
+
+  def _GetFinalizationKey(self):
+    """Generates the finalized key for the store.
+
+    Returns:
+      str: Redis key for the the finalization flag.
+    """
+    return '{0:s}-{1:s}'.format(
+        self._session_identifier, self._FINALIZED_KEY_NAME)
+
+  def _HasAttributeContainers(self, container_type):
+    """Determines if the store contains a specific type of attribute container.
+
+    Args:
+      container_type (str): attribute container type.
+
+    Returns:
+      bool: True if the store contains the specified type of attribute
+          containers.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    number_of_containers = self._redis_client.hlen(container_key)
+    return  number_of_containers > 0
+
+  def _RaiseIfNotReadable(self):
+    """Checks that the store is ready to for reading.
+
+     Raises:
+       IOError: if the store cannot be read from.
+       OSError: if the store cannot be read from.
+    """
+    if not self._redis_client:
+      raise IOError(
+          'Unable to read, client not connected.')
+
+  def _RaiseIfNotWritable(self):
+    """Checks that the store is ready to for writing.
+
+     Raises:
+       IOError: if the store cannot be written to.
+       OSError: if the store cannot be written to.
+    """
+    if not self._redis_client:
+      raise IOError('Unable to write, client not connected.')
+
+  def _WriteStorageMetadata(self):
+    """Writes the storage metadata."""
+    metadata = {
+        'format_version': self._FORMAT_VERSION,
+        'storage_type': definitions.STORAGE_TYPE_TASK,
+        'serialization_format': self.serialization_format}
+    metadata_key = self._GenerateRedisKey('metadata')
+
+    self._redis_client.hmset(metadata_key, metadata)
+
+  def _WriteAttributeContainer(self, attribute_container):
+    """Writes an attribute container to the store.
+
+    Args:
+      attribute_container (AttributeContainer): attribute container.
+    """
+    container_type = attribute_container.CONTAINER_TYPE
+    self._AddAttributeContainer(container_type, attribute_container)
+
+  def AddEvent(self, event, serialized_data=None):
+    """Adds an event.
+
+    Args:
+      event (EventObject): event.
+      serialized_data (Optional[bytes]): serialized form of the event.
+    """
+    super(RedisStore, self).AddEvent(event, serialized_data=serialized_data)
+    event_index_name = self._GenerateRedisKey(self._EVENT_INDEX_NAME)
+    identifier = event.GetIdentifier()
+    string_identifier = identifier.CopyToString()
+    self._redis_client.zincrby(
+        event_index_name, event.timestamp, string_identifier)
+
+  def Close(self):
+    """Closes the store."""
+    self._redis_client = None
+
+  def RemoveAttributeContainer(self, container_type, identifier):
+    """Removes an attribute container from the store.
+
+    Args:
+      container_type (Type): container type attribute of the container being
```

Comment:
`Type` should likely be `str`, can you check the rest of your changes for this as well

Reply:
Done